common/djangoapps/third_party_auth/api/views.py

```diff
@@ -20,13 +24,129 @@
+        """
+        if view.is_unprivileged_query(request, view.get_identifier_for_requested_user(request)):
+            return super(ProviderBaseThrottle, self).allow_request(request, view)
+        return True
+
+
+class ProviderBurstThrottle(ProviderBaseThrottle):
+    """
+    Maximum number of provider requests in a quick burst.
+    """
+    rate = settings.TPA_PROVIDER_BURST_THROTTLE  # Default '10/min'
+
+
+class ProviderSustainedThrottle(ProviderBaseThrottle):
+    """
+    Maximum number of provider requests over time.
+    """
+    rate = settings.TPA_PROVIDER_SUSTAINED_THROTTLE  # Default '50/day'
+
+
+class BaseUserView(APIView):
+    """
+    Common core of UserView and UserViewV2
+    """
+    identifier = namedtuple('identifier', ['kind', 'value'])
+    identifier_kinds = ['email', 'username']
+
+    authentication_classes = (
+        # Users may want to view/edit the providers used for authentication before they've
+        # activated their account, so we allow inactive users.
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    throttle_classes = [ProviderSustainedThrottle, ProviderBurstThrottle]
+
+    def do_get(self, request, identifier):
+        """
+        Fulfill the request, now that the identifier has been specified.
+        """
+        is_unprivileged = self.is_unprivileged_query(request, identifier)
+
+        if is_unprivileged:
+            if not getattr(settings, 'ALLOW_UNPRIVILEGED_SSO_PROVIDER_QUERY', False):
+                return Response(status=status.HTTP_403_FORBIDDEN)
+        try:
+            user = User.objects.get(**{identifier.kind: identifier.value})
+        except User.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        providers = pipeline.get_provider_user_states(user)
+
+        active_providers = [
+            self.get_provider_data(assoc, is_unprivileged)
+            for assoc in providers if assoc.has_account
+        ]
+
+        # In the future this can be trivially modified to return the inactive/disconnected providers as well.
+
+        return Response({
+            "active": active_providers
+        })
+
+    def get_provider_data(self, assoc, is_unprivileged):
+        """
+        Return the data for the specified provider.
+
+        If the request is unprivileged, do not return the remote ID of the user.
+        """
+        provider_data = {
+            "provider_id": assoc.provider.provider_id,
+            "name": assoc.provider.name,
+        }
+        if not is_unprivileged:
+            provider_data["remote_id"] = assoc.remote_id
+        return provider_data
+
+    def is_unprivileged_query(self, request, identifier):
+        """
+        Return True if a non-superuser requests information about another user.
+
+        Params must be a dict that includes only one of 'username' or 'email'
+        """
+        assert identifier.kind in {'username', 'email'}
```

Comment:
Any reason not to use `self.identifier_kinds` here? Could you also change this to raise a ValueError instead of using an `assert`?

Reply:
@jcdyer I'm still seeing a conflict here. Can you take one more look?

Reply:
You are correct.  The rebase didn't take properly.  Sorry about that!